### DISCUSSION TITLE
SPM modules

### DISCUSSION THREAD

**Discussion by ph-qyn** [brainvisa/brainvisa.github.io#30] [General]
Hi BrainVISA team

Thanks a lot for the great tool !
I would like to reproduce a referenced method that I use with SPM8 standalone within BrainVISA (4.6.1).
Starting from a T1 MRI, I use the following SPM modules
-   Segment
-   Normalise : Write
The deformation field output of Segment, a .mat file, is given as an input parameter to the Normalise module.

As I understood, the Normalise : write of SPM8 is available in Brainvisa whereas only the New Segment module is implemented which produces a .nii output for the deformation field. Therefore it cannot be given as an input to Normalise : Write only.
My question is : is there a way to combine Segment and Normalise : Write only modules ?
Either by :
  -  using the "old" segment function of SPM8 in brainVISA. If it is not implemented, will it be in future realeases ?
  -  converting the .nii into a format readable by the normalise function ?

Thank you in advance,

Pauline

**Comment by Hboni** [comment]
Hello Pauline

Thanks for using BrainVISA, and for the first discussion on this new version of the BrainVISA forum!

For now, as you mentioned, the SPM8 Segment process is not implemented in the SPM toolbox of BrainVISA. Its implementation is not planned because we did not have the need for it, we add integration of SPM12 processes mainly.

It is possible to use this deformation field generated by `SPM8 New Segment` to deform your data, using a process in SPM8, in utils toolbox, there is a `Deformation - apply deformation field`. But results can be slightly different from using `SPM8 Normalize`with `.mat` file from `SPM8 Segment`.

To solve your problem:

- We can implement `SPM8 Segment` in BrainVISA to have the exactly same processing for your data as you expect.
- You can use `SPM8 New Segment` with `Deformation - apply deformation field`

If finally you really need to use `SPM8 Segment` instead of `SPM8 New Segment`, we can plan to implement this process in BrainVISA.

Hugo

**Comment by ph-qyn** [comment]
Hello Hugo,

Thanks a lot for the very clear answer. I'll keep for now the New Segment + apply deformation field approach.

Pauline

**Reply by Hboni** [reply]
Hello Pauline,

No problem, do not hesitate to ask if you have other questions or if you want some features to be added to Brainvisa. :+1:

Hugo
